### PR TITLE
Add csi plugin in e2e test

### PR DESCRIPTION
**Pull request by danfengliu**
Thank you for contributing to Velero!

# Please add a summary of your change
Fixed issue: #4765
# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by blackpiglet — DISMISSED** [review]

**Review by blackpiglet — DISMISSED** [review]

**Review by blackpiglet — APPROVED** [review]

File: test/e2e/e2e_suite_test.go [hunk 1]
```diff
@@ -55,24 +55,25 @@ func init() {
...
 }
 
 var _ = Describe("[APIGroup] Velero tests with various CRD API group versions", APIGropuVersionsTest)
 
 // Test backup and restore of Kibishi using restic
-var _ = Describe("[Restic] Velero tests on cluster using the plugin provider for object storage and Restic for volume backups", BackupRestoreWithRestic)
+var _ = Describe("[Basic][Restic] Velero tests on cluster using the plugin provider for object storage and Restic for volume backups", BackupRestoreWithRestic)
 
-var _ = Describe("[Snapshot] Velero tests on cluster using the plugin provider for object storage and snapshots for volume backups", BackupRestoreWithSnapshots)
+var _ = Describe("[Basic][Snapshot] Velero tests on cluster using the plugin provider for object storage and snapshots for volume backups", BackupRestoreWithSnapshots)
```
Comment (ywk253100) [inline test/e2e/e2e_suite_test.go:74]: The `Restic` and `Snapshot` cases cannot run on KinD, so don't label them as `Basic`

File: test/e2e/util/csi/AzureVolumeSnapshotClass.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotClass
+metadata:
+  name: velero
+  labels:
+    velero.io/csi-volumesnapshot-class: "true"
+driver: disk.csi.azure.com
+deletionPolicy: Retain
```
Comment (ywk253100) [inline test/e2e/util/csi/AzureVolumeSnapshotClass.yaml:8]: Change the policy to `Delete`?

File: test/e2e/e2e_suite_test.go (hunk 1 above)
Comment (danfengliu) [reply test/e2e/e2e_suite_test.go:74]: In this PR,  I have modified GINKGO_FOCUS='Basic\].+\[ClusterResource' .

File: test/e2e/e2e_suite_test.go [hunk 3]
```diff
@@ -55,24 +55,25 @@ func init() {
 	flag.StringVar(&VeleroCfg.VeleroNamespace, "velero-namespace", "velero", "namespace to install Velero into")
 	flag.BoolVar(&VeleroCfg.InstallVelero, "install-velero", true, "install/uninstall velero during the test.  Optional.")
 	flag.StringVar(&VeleroCfg.RegistryCredentialFile, "registry-credential-file", "", "file containing credential for the image registry, follows the same format rules as the ~/.docker/config.json file. Optional.")
-	flag.StringVar(&VeleroCfg.KibishiiDirectory, "kibishii-directory", "github.com/vmware-tanzu-experiments/distributed-data-generator/kubernetes/yaml/", "The file directory or URL path to install Kibishii. Optional.")
-
+	flag.StringVar(&VeleroCfg.KibishiiDirectory, "kibishii-directory", "github.com/danfengliu/distributed-data-generator/kubernetes/yaml/", "The file directory or URL path to install Kibishii. Optional.")
```
[the remark this answers is not in the document]
Comment (danfengliu) [reply test/e2e/e2e_suite_test.go:58]: Yes, I forgot this, thanks.

File: test/e2e/e2e_suite_test.go [hunk 4]
```diff
@@ -55,24 +55,25 @@ func init() {
...
 	flag.StringVar(&VeleroCfg.AdditionalBSLProvider, "additional-bsl-object-store-provider", "", "Provider of object store plugin for additional backup storage location. Required if testing multiple credentials support.")
 	flag.StringVar(&VeleroCfg.AdditionalBSLBucket, "additional-bsl-bucket", "", "name of the object storage bucket for additional backup storage location. Required if testing multiple credentials support.")
 	flag.StringVar(&VeleroCfg.AdditionalBSLPrefix, "additional-bsl-prefix", "", "prefix under which all Velero data should be stored within the bucket for additional backup storage location. Optional.")
 	flag.StringVar(&VeleroCfg.AdditionalBSLConfig, "additional-bsl-config", "", "configuration to use for the additional backup storage location. Format is key1=value1,key2=value2")
 	flag.StringVar(&VeleroCfg.AdditionalBSLCredentials, "additional-bsl-credentials-file", "", "file containing credentials for additional backup storage location provider. Required if testing multiple credentials support.")
+	flag.StringVar(&VeleroCfg.Features, "features", "", "Comma-separated list of features to enable for this Velero process.")
```
Comment (ywk253100) [inline test/e2e/e2e_suite_test.go:66]: Seems that the CSI and non-CSI snapshot cases use the same code, they are differentiated by different inputs?

But I cannot run one single make command to run all the end-2-end test cases including both CSI and non-CSI snapshot cases

**Review by ywk253100 — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Pull request merged**